### PR TITLE
capstone: fix pkgconfig file generation

### PR DESCRIPTION
**Pull request by joepie91**
###### Motivation for this change

The generation of pkgconfig `.pc` files is currently broken, because `pkgconfig` isn't being passed to the expression as a dependency. This leads to failures when attempting to build other software that relies on capstone through pkgconfig (eg. `edb`).

This PR fixes that by adding `pkgconfig` as a dependency. The buildscript picks it up automatically and correctly generates a `.pc` file (which is correctly picked up by other expressions).

Note that while I've tested this patch locally, I've tested it differently (namely, by copying the expression to a custom repository and modifying it). I see no reason why it wouldn't work as submitted in this PR, but figured I'd make note of this anyway.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/capstone/default.nix [hunk 1]
```diff
@@ -12,6 +12,10 @@ stdenv.mkDerivation rec {
   configurePhase = '' patchShebangs make.sh '';
   buildPhase = '' ./make.sh '';
   installPhase = '' env PREFIX=$out ./make.sh install '';
+  
+  buildInputs = [
+    pkgconfig
```
Comment (Mic92) [inline pkgs/development/libraries/capstone/default.nix]: This should be in `nativeBuildInputs`:

```
nativeBuildInputs = [ pkgconfig ];
```

Comment (joepie91) [inline pkgs/development/libraries/capstone/default.nix]: I can't say I really fully understand the difference between `buildInputs` and `nativeBuildInputs`. Could you elaborate?

Comment (Mic92) [inline pkgs/development/libraries/capstone/default.nix]: This is important for cross-compilation. All build-time dependencies should be specified in `nativeBuildInputs`.
For example host compiler would be in `nativeBuildInputs`, while a library linked against the program would be in `buildInputs`.

**Review by zimbatm — APPROVED** [review]

File: pkgs/development/libraries/capstone/default.nix (hunk 1 above)
Comment (joepie91) [inline pkgs/development/libraries/capstone/default.nix]: Ah, I see. Thanks for the explanation :)

**Pull request merged**